test/governance/InterchainMultisig.js

```diff
@@ -0,0 +1,333 @@
+const chai = require('chai');
+const { ethers, network } = require('hardhat');
+const { sortBy } = require('lodash');
+const { getAddresses, encodeInterchainCallsBatch, getWeightedSignaturesProof } = require('../utils');
+const {
+    utils: { formatBytes32String },
+} = ethers;
+const { expect } = chai;
+
+describe('InterchainMultisig', () => {
+    const threshold = 2;
+    const nativeValue = 100;
+
+    let wallets;
+    let owner;
+    let signers;
+    let newSigners = [];
+
+    let interchainMultisigFactory;
+    let interchainMultisig;
+    let targetFactory;
+    let targetContract;
+    let calldata;
+
+    before(async () => {
+        wallets = await ethers.getSigners();
+
+        owner = wallets[0];
+        signers = sortBy(wallets.slice(1, 3), (wallet) => wallet.address.toLowerCase());
+        newSigners = sortBy(wallets.slice(0, 2), (wallet) => wallet.address.toLowerCase());
+
+        interchainMultisigFactory = await ethers.getContractFactory('InterchainMultisig', owner);
+        targetFactory = await ethers.getContractFactory('Target', owner);
+
+        // new multisig and target contracts for each test
+        interchainMultisig = await interchainMultisigFactory.deploy('Ethereum', [
+            getAddresses(signers),
+            signers.map(() => 1),
+            threshold,
+        ]);
+
+        await interchainMultisig.deployTransaction.wait(network.config.confirmations);
+
+        targetContract = await targetFactory.deploy().then((d) => d.deployed());
+
+        calldata = targetContract.interface.encodeFunctionData('callTarget');
+    });
+
+    it('should validate storage constants', async () => {
+        const testMultisigFactory = await ethers.getContractFactory('TestInterchainMultisig', owner);
+
+        await testMultisigFactory.deploy('Ethereum', [getAddresses(signers), signers.map(() => 1), threshold]);
+    });
+
+    it('should revert on execute with insufficient value sent', async () => {
+        const call = ['Ethereum', interchainMultisig.address, targetContract.address, calldata, nativeValue];
+
+        await expect(
+            interchainMultisig.executeCalls(
+                formatBytes32String('5'),
+                [call],
+                getWeightedSignaturesProof(
+                    encodeInterchainCallsBatch(formatBytes32String('5'), [call]),
+                    signers,
+                    signers.map(() => 1),
+                    2,
+                    signers,
+                ),
+            ),
+        ).to.be.revertedWithCustomError(interchainMultisig, 'InsufficientBalance');
```

Comment:
Use `expertRevert` instead for live testnet compatibility, applies to other instances of `revertedWithCustomError` as well